api/testharness/harness.go

```diff
@@ -285,6 +290,122 @@ func (h *Harness) TestTracer(subjectFactory func() trace.Tracer) {
 			e.Expect(csc.SpanID).NotToEqual(psc.SpanID)
 		})
 	})
+
+	h.testSpan(subjectFactory)
+}
+
+func (h *Harness) testSpan(tracerFactory func() trace.Tracer) {
+	var methods = map[string]func(span trace.Span){
+		"#End": func(span trace.Span) {
+			span.End()
+		},
+		"#AddEvent": func(span trace.Span) {
+			span.AddEvent(context.Background(), "test event")
+		},
+		"#AddEventWithTimestamp": func(span trace.Span) {
+			span.AddEventWithTimestamp(context.Background(), time.Now(), "test event")
+		},
+		"#AddLink": func(span trace.Span) {
+			span.AddLink(trace.Link{})
+		},
+		"#Link": func(span trace.Span) {
+			span.Link(core.SpanContext{})
+		},
+		"#SetStatus": func(span trace.Span) {
+			span.SetStatus(codes.Internal)
+		},
+		"#SetName": func(span trace.Span) {
+			span.SetName("new name")
+		},
+		"#SetAttribute": func(span trace.Span) {
+			span.SetAttribute(core.Key("key").String("value"))
+		},
+		"#SetAttributes": func(span trace.Span) {
+			span.SetAttributes(core.Key("key1").String("value"), core.Key("key2").Int(123))
```

Comment:
Unrelated nit:  you can now write `key.String("key1", "value")` instead of `core.Key("key1").String("value")`.

Reply:
That's from the "go.opentelemetry.io/api/key" package, just for the record.